Source/WebKit/UIProcess/Extensions/Cocoa/WebExtensionActionCocoa.mm

```diff
@@ -629,11 +643,14 @@ - (void)_otherPopoverWillShow:(NSNotification *)notification
         return nullptr;
 
     // Tab actions fallback to the window action.
-    if (m_tab)
-        return extensionContext()->getAction(m_tab->window().get()).ptr();
+    if (RefPtr tab = this->tab())
+        return extensionContext()->getAction(tab->window().get()).ptr();
+    // Tab actions whose tab references have dropped fall back to the default action
+    if (m_tab.has_value())
```

Comment:
```suggestion
        return extensionContext()->getAction(tab->window().get()).ptr();

    // Tab actions whose tab references have dropped fallback to the default action.
    if (m_tab.has_value())
```

Could also group with the window fallback below:

```c++
// Window actions and tab actions whose tab references have dropped fallback to the default action.
if (m_window.has_value() || m_tab.has_value())

Reply:
I'll group it with the window fallback, feels like that's probably what I should have done here in the first place